src/Libraries/Microsoft.Extensions.Telemetry/Logging/ExtendedLogger.LegacyTagJoiner.cs

```diff
@@ -76,5 +76,7 @@ public void SetIncomingTags(IReadOnlyList<KeyValuePair<string, object?>> value)
         }
 
         IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
+        public override string? ToString() => State?.ToString();
```

Comment:
As I understand, in case of the modern logging path (the tag joiner is a ModernTagJoiner, and the State is a LoggerMessageState) we'll output the LoggerMessageState serialized to a string as the Message, right? Is it the desired behavior?

Reply:
Yes, that is correct.

The LegacyTagJoiner deals with log state generated using the old LogInformation/LogDebug/LogError methods. For those cases, the TState comes in as a string which is meaningful to apply ToString on.

But in the ModernTagJoiner, the TState is a LoggerMessageState instance and there isn't a string available there.